src/mavsdk/plugins/action/action_impl.cpp

```diff
@@ -460,11 +465,15 @@ void ActionImpl::goto_location_async(
                     command_result_callback(result, callback);
                 });
         };
-
-    // Change to Hold mode first
-    if (_system_impl->get_flight_mode() != FlightMode::Hold) {
+    // If PX4 change to Hold mode first
+    FlightMode gotoFlightMode = FlightMode::Hold;
+    // If ArduPilot change to Offboard mode first
+    if (_system_impl->autopilot() != SystemImpl::Autopilot::Px4) {
+        gotoFlightMode = FlightMode::Offboard;
+    }
```

Comment:
Hmm I think it would read better like this:

```suggestion
    FlightMode goto_flight_mode;
    if (_system_impl->autopilot() == SystemImpl::Autopilot::Px4) {
        goto_flight_mode = FlightMode::Hold;
    } else {
        gotoFlightMode = FlightMode::Offboard;
    }
```

@paulholmstedt: What do you think? I won't ask you to resend a patch with those changes, I can do it myself if you agree with the suggestion :blush:.

Reply:
Agreed.

Reply:
Small variable case fixup:

```suggestion
    FlightMode goto_flight_mode;
    if (_system_impl->autopilot() == SystemImpl::Autopilot::Px4) {
        goto_flight_mode = FlightMode::Hold;
    } else {
        goto_flight_mode = FlightMode::Offboard;
    }
```